### PR TITLE
Better instructions to 'add resources to a task'

### PR DESCRIPTION
**Pull request by SmritiSatyanV**
Slack conversation: https://flyte-org.slack.com/archives/C02HARLSW30/p1653436394626289
Improve the documentation for adding resources (such as memory, cpu) that a task can use in Flyte.
Signed-off-by: SmritiSatyanV <smriti@union.ai>

### CODE REVIEW

File: rsts/deployment/cluster_config/general.rst [hunk 1]
```diff
@@ -7,24 +7,24 @@ Configuring Custom K8s Resources
...
 ***************************
 
-Many platform specifications such as task resource defaults, project namespace Kubernetes quota, and more can all be
+Many platform specifications such as task resource defaults, project namespace Kubernetes quota, and more can be
 assigned using default values or custom overrides. Defaults are specified in the FlyteAdmin config and
-overrides for specific projects can be registered with the FlyteAdmin service.
+overrides for specific projects are registered with the FlyteAdmin service.
 
-Specifically, flyte allows these custom settings along increasing levels of specificity:
+Specifically, Flyte allows these custom settings along increasing levels of specificity:
```
Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:14]: Suggestion: "You can customize these settings along..."

File: rsts/deployment/cluster_config/general.rst [hunk 2]
```diff
@@ -38,7 +38,7 @@ in the FlyteAdmin config are used.
 
 The override values from the database are assigned at execution, rather than registration time.
```
Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:39]: Passive. Who or what assigns the override values from the database?

Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:39]: Passive. Who or what assigns the override values from the database?

File: rsts/deployment/cluster_config/general.rst [hunk 3]
```diff
@@ -75,16 +75,15 @@ These are free-form key-value pairs used when filling the templates that the adm
 
 The keys represent templatized variables in `cluster resource template YAML <https://github.com/flyteorg/flyte/blob/1e3d515550cb338c2edb3919d79c6fa1f0da5a19/charts/flyte-core/values.yaml#L737,L760>`__ and the values are what you want to see filled in.
 
-
 In the absence of custom override values, ``templateData`` from the `FlyteAdmin config <https://github.com/flyteorg/flyte/blob/1e3d515550cb338c2edb3919d79c6fa1f0da5a19/charts/flyte-core/values.yaml#L719,L734>`__
-is used as a default. Note, these defaults are specified by domain but will be applied to every project-domain namespace combination.
+is used as a default. These defaults are specified by domain but will be applied to every project-domain namespace combination.
```
Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:79]: Passive. Who or what uses this default?

Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:79]: Passive. Who or what uses this default?

Passive. Who or what specifies the defaults by domain?

File: rsts/deployment/cluster_config/general.rst [hunk 4]
```diff
@@ -75,16 +75,15 @@ These are free-form key-value pairs used when filling the templates that the adm
...
 
 .. note::
     The above-referenced settings can only take on domain, project, and domain specificity.
```
Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:82]: Verbose. Suggestion: "The settings above..."
 
Jargony.  "Specificity" is a big word. Is there a simpler, more direct word?

Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:82]: Verbose. Suggestion: "The settings above..."
 
Jargony.  "Specificity" is a big word. Is there a simpler, more direct word?

File: rsts/deployment/cluster_config/general.rst [hunk 5]
```diff
@@ -75,16 +75,15 @@ These are free-form key-value pairs used when filling the templates that the adm
...
-    Since Flyte has not tied in the notion of a workflow or a launch plan to any Kubernetes constructs, specifying a workflow or launch plan name doesn't make any sense.
+    Since Flyte is not tied in the notion of a workflow or a launch plan to any Kubernetes constructs, specifying a workflow or launch plan name doesn't make sense.
```
Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:83]: "tied to the notion..."

Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:83]: "tied to the notion..."

File: rsts/deployment/cluster_config/general.rst [hunk 6]
```diff
@@ -154,12 +153,11 @@ You can view all custom execution cluster attributes by visiting ``protocol://<h
 
 Execution Queues
 ================
-Execution queues themselves are currently defined in the
+Execution queues are currently defined in
```
Comment (marc-union) [inline rsts/deployment/cluster_config/general.rst:156]: Delete "currently"

**Review by marc-union — COMMENTED** [review]
Nice work. Let me know if you have questions.

File: rsts/deployment/cluster_config/general.rst [hunk 7]
```diff
@@ -94,7 +92,7 @@ Define an attributes file like the the following in ``cra.yaml``
     domain: development
     project: flyteexamples
 
-Then run the below command to ensure that when the admin fills in cluster resource templates, the Kubernetes namespace ``flyteexamples-development`` will have a resource quota of 1000 CPU cores and 5TB of memory.
+To ensure that when the admin fills in cluster resource templates, the Kubernetes namespace ``flyteexamples-development`` will have a resource quota of 1000 CPU cores and 5TB of memory:
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:95]:
```suggestion
To ensure that when the admin fills in cluster resource templates, the Kubernetes namespace ``flyteexamples-development`` will have a resource quota of 1000 CPU cores and 5TB of memory, run:
```

File: rsts/deployment/cluster_config/general.rst [hunk 8]
```diff
@@ -36,9 +36,9 @@ In the absence of an override, the global
 `default values <https://github.com/flyteorg/flyte/blob/1e3d515550cb338c2edb3919d79c6fa1f0da5a19/charts/flyte-core/values.yaml#L35,L43>`__
 in the FlyteAdmin config are used.
 
-The override values from the database are assigned at execution, rather than registration time.
+You can override values from the database, that are assigned at execution, rather than registration time.
 
-To update individual project-domain attributes, first define a ``tra.yaml`` file with your overrides:
+To customize resources for project-domain attributes, define a ``tra.yaml`` file with the values you need:
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:41]:
```suggestion
To customize resources for project-domain attributes, define a ``tra.yaml`` file with your overrides:
```

File: rsts/deployment/cluster_config/general.rst [hunk 9]
```diff
@@ -36,9 +36,9 @@ In the absence of an override, the global
 `default values <https://github.com/flyteorg/flyte/blob/1e3d515550cb338c2edb3919d79c6fa1f0da5a19/charts/flyte-core/values.yaml#L35,L43>`__
 in the FlyteAdmin config are used.
 
-The override values from the database are assigned at execution, rather than registration time.
+You can override values from the database, that are assigned at execution, rather than registration time.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:39]: This is wrong. Can you revert?

File: rsts/deployment/cluster_config/general.rst [hunk 10]
```diff
@@ -51,15 +51,15 @@ To update individual project-domain attributes, first define a ``tra.yaml`` file
     project: flyteexamples
     domain: development
 
-Then, use the following command for your reference.
+Update the task resource attributes for a project-domain-workflow combination:
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:54]: Why is it `project-domain-workflow`? Isn't it `project-domain`?

File: rsts/deployment/cluster_config/general.rst [hunk 11]
```diff
@@ -103,21 +101,21 @@ Then run the below command to ensure that when the admin fills in cluster resour
 Refer to the :ref:`docs <flytectl:flytectl_update_cluster-resource-attribute>` to learn more about the command and its supported flag(s).
 
 
-To fetch and verify the individual project-domain attributes, similarly you can run:
+To fetch and verify the individual project-domain attributes:
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:104]:
```suggestion
To fetch and verify the individual project-domain attributes, run the following command:
```

File: rsts/deployment/cluster_config/general.rst [hunk 12]
```diff
@@ -103,21 +101,21 @@ Then run the below command to ensure that when the admin fills in cluster resour
...
 
 .. prompt:: bash
 
     flytectl get cluster-resource-attribute -p flyteexamples -d development
 
 Refer to the :ref:`docs <flytectl:flytectl_get_task-resource-attribute>` to learn more about the command and its supported flag(s).
 
-The above-updated values will, in turn, be used to fill in the template fields for the flyteexamples-development namespace.
+You can use these updated values to fill the template fields for the flyteexamples-development namespace.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:112]: A user doesn't fill in the template fields; it's done by Flyte. I think it's ok to use passive voice in these scenarios.

File: rsts/deployment/cluster_config/general.rst [hunk 13]
```diff
@@ -103,21 +101,21 @@ Then run the below command to ensure that when the admin fills in cluster resour
...
 
 For other namespaces, the `platform defaults <https://github.com/flyteorg/flyte/blob/1e3d515550cb338c2edb3919d79c6fa1f0da5a19/charts/flyte-core/values.yaml#L719,L734>`__ apply.
 
 .. note::
-    The template values, e.g. ``projectQuotaCpu`` or ``projectQuotaMemory`` are freeform strings.
-    You must ensure that they match the template placeholders in your `template file <https://github.com/flyteorg/flyte/blob/master/kustomize/base/single_cluster/headless/config/clusterresource-templates/ab_project-resource-quota.yaml>`__
+    The template values, for example: ``projectQuotaCpu`` or ``projectQuotaMemory`` are free-form strings.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:117]:
```suggestion
    The template values, for example, ``projectQuotaCpu`` or ``projectQuotaMemory`` are free-form strings.
```
I don't think you should use a colon (`:`) when using "for example" in a sentence.

File: rsts/deployment/cluster_config/general.rst [hunk 14]
```diff
@@ -126,23 +124,23 @@ Execution Cluster Label
...
+Ensure that admin places executions in the flyteexamples project and development domain onto ``mycluster``:
 
 .. prompt:: bash
 
    flytectl update execution-cluster-label --attrFile ec.yaml
 
 Refer to the :ref:`docs <flytectl:flytectl_update_execution-cluster-label>` to learn more about the command and its supported flag(s).
 
-To fetch and verify the individual project-domain attributes, similarly you can run:
+To fetch and verify the individual project-domain attributes:
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/general.rst:143]:
```suggestion
To fetch and verify the individual project-domain attributes, run:
```
I feel "run" is necessary to tell the user to run the command. But if it's not necessary, feel free to remove it.

File: rsts/deployment/cluster_config/general.rst (hunk 12 above)
Comment (SmritiSatyanV) [reply rsts/deployment/cluster_config/general.rst:112]: "Flyte uses these updated values to fill the template fields for the flyteexamples-development namespace." sounds better IMO.

File: rsts/deployment/cluster_config/general.rst (hunk 14 above)
Comment (SmritiSatyanV) [reply rsts/deployment/cluster_config/general.rst:143]: The 'run' somehow feels extra. We can keep it like this.

**Review by samhita-alla — APPROVED** [review]

**Pull request merged**